pkg/database/database.go

```diff
@@ -120,12 +126,37 @@ func (db *Database) Open(ctx context.Context) error {
 	return db.OpenWithCacher(ctx, nil)
 }
 
+const driverName = "ocsql"
+
+func stringInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+
 // OpenWithCacher creates a database connection with the cacher. This should
 // only be called once.
 func (db *Database) OpenWithCacher(ctx context.Context, cacher cache.Cacher) error {
 	c := db.config
 
-	rawDB, err := gorm.Open("postgres", c.ConnectionString())
+	driver := ocsql.Wrap(&postgres.Driver{}, ocsql.WithAllTraceOptions())
```

Comment:
Two things here:

1. This feels like a lot of code, and I'm not entirely sure why it's all necessary. Can we just register the callback?

1. I'm concerned that this observability might allow a server operator to correlate specific database events (e.g. an insert) which a TEK upload. It's the same problem as the network observer. Since our chaff requests don't hit the database, it would be easy to separate chaff from non-chaff /cc @mikehelmick

Reply:
Hi Seth,

About your first point, the callbacks will actually emit only gorm specific metrics. https://github.com/sagikazarmark/go-gin-gorm-opencensus/blob/master/pkg/ocgorm/stats.go. 

For postgres specific metrics, we need to override the driver. Following are the metrics that are emitted by https://github.com/opencensus-integrations/ocsql#recorded-metrics.

For point 2, I think we should disable tracing in that case and that should prevent the server operator to correlate anything.

Reply:
I have changed the code to remove traceability. WDYT?

Reply:
I think we're gonna need to hold off until we hear from the privacy team (I pinged them).

Reply:
Also worth remembering, we're sampling traces at 0.4%.

Reply:
@sethvargo any update from the privacy team?

Reply:
@laurentsimon ^